skrub/tests/test_table_vectorizer.py

```diff
@@ -307,121 +326,147 @@ def passthrough_vectorizer():
     )
 
 
-@pytest.mark.parametrize("X, dict_expected_types", X_tuples)
-def test_auto_cast(X, dict_expected_types):
+def test_auto_cast(X_tuples_fixture):
```

Comment:
I would have found it markedly more readable to use "@pytest.mark.parametrize" (https://docs.pytest.org/en/stable/how-to/parametrize.html#parametrize-basics) instead of the fixture.

Is there a reason that you did not consider this option?